pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,104 @@
+package scheduler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/graph"
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+func NewScheduler(l hclog.Logger, c CapacityProvider, url string) *Scheduler {
+	x := Scheduler{
+		l:                l.Named("scheduler"),
+		capacityProvider: c,
+		apiClient:        graph.NewAPIClient(l),
+		queue:            make([]Build, 0),
+		queueMutex:       new(sync.Mutex),
+		tuples:           make([]types.SpecTuple, 0),
+	}
+	x.apiClient.Url = url
+
+	return &x
+}
+
+// Pops a build off the queue and hands it off to the CapacityProvider.
+func (s *Scheduler) send() bool {
+	s.queueMutex.Lock()
+	defer s.queueMutex.Unlock()
+
+	if len(s.queue) == 0 {
+		return false
+	}
+	err := s.capacityProvider.DispatchBuild(s.queue[0])
```

Comment:
This line and the below if statement should be combined `if err := s.capacityProvider.DispatchBuild(...); err != nil {`